crates/settlement-clients/ethereum/src/tests/mod.rs

```diff
@@ -0,0 +1,318 @@
+    assert!(txn.signature.is_some());
+    assert_eq!(txn.to.unwrap().to_string(), *TEST_DUMMY_CONTRACT_ADDRESS);
+
+    // Testing verify_tx_inclusion
+    sleep(Duration::from_secs(2)).await;
+    ethereum_settlement_client
+        .wait_for_tx_finality(update_state_result.as_str())
+        .await
+        .expect("Could not wait for txn finality.");
+    let verified_inclusion = ethereum_settlement_client
+        .verify_tx_inclusion(update_state_result.as_str())
+        .await
+        .expect("Could not verify inclusion.");
+    assert_eq!(verified_inclusion, SettlementVerificationStatus::Verified);
+}
+
+#[rstest]
+#[tokio::test]
+#[case::basic(20468828)]
+async fn update_state_blob_with_impersonation_works(#[case] block_no: u64) {
+    let TestFixture { ethereum_settlement_client, provider } = ethereum_test_fixture(block_no);
+
+    provider.anvil_impersonate_account(*STARKNET_OPERATOR_ADDRESS).await.expect("Unable to impersonate account.");
+
+    let nonce = ethereum_settlement_client.get_nonce().await.expect("Unable to fetch nonce");
+
+    // Create a contract instance.
+    let contract = STARKNET_CORE_CONTRACT::new(*STARKNET_CORE_CONTRACT_ADDRESS, provider.clone());
+
+    // Call the contract, retrieve the current stateBlockNumber.
+    let prev_block_number = contract.stateBlockNumber().call().await.unwrap();
+
+    // generating program output and blob vector
+    let program_output = get_program_output(block_no);
+    let blob_data_vec = get_blob_data(block_no);
+
+    // Calling update_state_with_blobs
+    let update_state_result = ethereum_settlement_client
+        .update_state_with_blobs(program_output, blob_data_vec, nonce)
+        .await
+        .expect("Could not go through update_state_with_blobs.");
+
+    // Asserting, Expected to receive transaction hash.
+    assert!(!update_state_result.is_empty(), "No transaction Hash received.");
+
+    // Call the contract, retrieve the latest stateBlockNumber.
+    let latest_block_number = contract.stateBlockNumber().call().await.unwrap();
+
+    println!("PREVIOUS BLOCK NUMBER {}", prev_block_number._0);
+    println!("CURRENT BLOCK HASH {}", latest_block_number._0);
+    assert_eq!(prev_block_number._0.as_u32() + 1, latest_block_number._0.as_u32());
+
+    // Testing verify_tx_inclusion
+    sleep(Duration::from_secs(2)).await;
+    ethereum_settlement_client
+        .wait_for_tx_finality(update_state_result.as_str())
+        .await
+        .expect("Could not wait for txn finality.");
+    let verified_inclusion = ethereum_settlement_client
+        .verify_tx_inclusion(update_state_result.as_str())
+        .await
+        .expect("Could not verify inclusion.");
+    assert_eq!(verified_inclusion, SettlementVerificationStatus::Verified);
+}
+
+#[rstest]
+#[tokio::test]
+#[case::typical(20468828, 666039)]
+async fn get_last_settled_block_typical_works(#[case] block_no: u64, #[case] expected: u64) {
```

Comment:
```suggestion
async fn get_last_settled_block_typical_works(#[case] fork_block_no: u64, #[case] expected: u64) {
```

Reply:
maybe we can do this for other cases as well

Reply:
Valid, implemented.